test/e2e/framework/predicates/conditions.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package predicates
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// objectWithStatus is a simplified struct to parse the status of a resource.
+type objectWithStatus struct {
+	Status struct {
+		Conditions []metav1.Condition `json:"conditions,omitempty"`
+	} `json:"status"`
+}
+
+// ReadyConditionIsTrue checks if the given object has a Ready condition set to True.
+func ReadyConditionIsTrue(obj client.Object) error {
+	u, err := asUnstructured(obj)
```

Comment:
I think mixing and matching unstructured and typed objects is going to be a gotcha - for example if someone tried to call this using `sandboxv1alpha1.Sandbox` they will likely get an error. The other function implementations all assume typed objects, so I'd suggest using the typed object for consistency here - but I can fix this with a follow up